lib/cylc/network/https/client_reporter.py

```diff
@@ -131,19 +131,22 @@ def signout(self, server_obj):
                 caller.user, caller.host, caller.prog_name, caller.uuid))
         try:
             del self.clients[caller.uuid]
-        except:
+        except KeyError:
             # Already forgotten.
             pass
         self._housekeep()
 
     def _housekeep(self):
         """Forget inactive clients."""
 
-        for uuid in self.clients.keys():
+        for uuid in self.clients.copy():
```

Comment:
In python 2 yes, in python 3 `dict.keys()` returns a view. I've changed this to `dict.items()` as my initial commit didn't fix the immediate problem.